haystack_experimental/components/agents/agent.py

```diff
@@ -238,10 +392,41 @@ def run(
             - "last_message": The last message exchanged during the agent's run.
             - Any additional keys defined in the `state_schema`.
         :raises RuntimeError: If the Agent component wasn't warmed up before calling `run()`.
+        :raises AgentBreakpointException: If an agent breakpoint is triggered.
+
         """
         if not self._is_warmed_up and hasattr(self.chat_generator, "warm_up"):
             raise RuntimeError("The component Agent wasn't warmed up. Run 'warm_up()' before calling 'run()'.")
 
+        if breakpoints and resume_state:
+            msg = (
+                "agent_breakpoint and resume_state cannot be provided at the same time. The agent run will be aborted."
+            )
+            raise ValueError(msg)
+
+        # validate breakpoints
+        if breakpoints:
+            breakpoints = self._validate_breakpoints(breakpoints)
+
+        # resume state if provided
+        if resume_state:
+            component_visits = resume_state.get("pipeline_state", {}).get("component_visits", {})
+            state_data = resume_state.get("pipeline_state", {}).get("inputs", {}).get("state", {}).get("data", {})
```

Comment:
I think `state_data` need to be deserialized. I believe @Amnah199 can help with this.